pkg/ccl/streamingccl/streamingest/stream_ingestion_job.go

```diff
@@ -274,6 +300,38 @@ func ingest(ctx context.Context, execCtx sql.JobExecContext, ingestionJob *jobs.
 	return errors.CombineErrors(ingestWithClient(), client.Close(ctx))
 }
 
+func ingestWithRetries(
+	ctx context.Context, execCtx sql.JobExecContext, ingestionJob *jobs.Job,
+) error {
+	ro := retry.Options{
+		InitialBackoff: 3 * time.Second,
+		Multiplier:     2,
+		MaxBackoff:     1 * time.Minute,
+		MaxRetries:     60,
+	}
+
+	var err error
+	retryCount := 0
+	for r := retry.Start(ro); r.Next(); {
+		err = ingest(ctx, execCtx, ingestionJob)
+		if err == nil {
+			break
+		}
+		if jobs.IsPermanentJobError(err) {
+			break
+		}
+		updateRunningStatus(ctx, ingestionJob,
+			fmt.Sprintf("stream ingestion waits for retrying after error %s", err))
```

Comment:
Assuming `updateRunningStatus` doesn't log anything, it'd be nice to log every time we're about to retry.